src/core/multi/qrcode/QRCodeMultiReader.ts

```diff
@@ -0,0 +1,193 @@
+
+  /**
+   * TYPESCRIPTPORT: this is an overloaded method so here it'll work only as a entrypoint for choosing which overload to call.
+   */
+  public decodeMultiple(image: BinaryBitmap, hints: Map<DecodeHintType, any> = null): Result[] {
+
+    if (hints && hints instanceof Map) {
+      return this.decodeMultipleImpl(image, hints);
+    }
+
+    return this.decodeMultipleOverload1(image);
+  }
+
+  /**
+   * @throws NotFoundException
+   * @override decodeMultiple
+   */
+  private decodeMultipleOverload1(image: BinaryBitmap): Result[] {
+    return this.decodeMultipleImpl(image, null);
+  }
+
+  /**
+   * @override
+   * @throws NotFoundException
+   */
+  private decodeMultipleImpl(image: BinaryBitmap, hints:  Map<DecodeHintType, any>): Result[] {
+    let results: List<Result> = [];
+    const detectorResults: DetectorResult[] = new MultiDetector(image.getBlackMatrix()).detectMulti(hints);
+    for (const detectorResult of detectorResults) {
+      try {
+        const decoderResult: DecoderResult = this.getDecoder().decodeBitMatrix(detectorResult.getBits(), hints);
+        const points: ResultPoint[] = detectorResult.getPoints();
+        // If the code was mirrored: swap the bottom-left and the top-right points.
+        if (decoderResult.getOther() instanceof QRCodeDecoderMetaData) {
+          (<QRCodeDecoderMetaData> decoderResult.getOther()).applyMirroredCorrection(points);
+        }
+        const result: Result = new Result(decoderResult.getText(), decoderResult.getRawBytes(), points,
+                                   BarcodeFormat.QR_CODE);
+        const byteSegments: List<Uint8Array> = decoderResult.getByteSegments();
+        if (byteSegments != null) {
+          result.putMetadata(ResultMetadataType.BYTE_SEGMENTS, byteSegments);
+        }
+        const ecLevel: string = decoderResult.getECLevel();
+        if (ecLevel != null) {
+          result.putMetadata(ResultMetadataType.ERROR_CORRECTION_LEVEL, ecLevel);
+        }
+        if (decoderResult.hasStructuredAppend()) {
+          result.putMetadata(ResultMetadataType.STRUCTURED_APPEND_SEQUENCE,
+                             decoderResult.getStructuredAppendSequenceNumber());
+          result.putMetadata(ResultMetadataType.STRUCTURED_APPEND_PARITY,
+                             decoderResult.getStructuredAppendParity());
+        }
+        results.push(result);
+      } catch (re) {
+        if (re instanceof ReaderException) {
+          // ignore and continue
+        } else {
+          throw re;
+        }
+      }
+    }
+    if (results.length === 0) {
+      return QRCodeMultiReader.EMPTY_RESULT_ARRAY;
+    } else {
+      results = QRCodeMultiReader.processStructuredAppend(results);
+      return results/* .toArray(QRCodeMultiReader.EMPTY_RESULT_ARRAY) */;
+    }
+  }
+
+  static processStructuredAppend( results: List<Result>): List<Result> {
+    const newResults: List<Result> = [];
+    const saResults: List<Result> = [];
```

Comment:
We need to create multiple `saResults: List<Result>` that grouped by `ResultMetadataType.STRUCTURED_APPEND_PARITY` to handle multi parity structured append qr codes in one document.

Reply:
Thanks for the tip!! Can you help me with some examples or a PR?